### ISSUE TITLE
WebGL2 support

### ISSUE THREAD

**Issue by dawidgorny** [tauri-apps/tauri#2866]
WebGL2 unavailable in the WebView.

### **To Reproduce**
1. `npx create-tauri-app`
2. Use source of `https://threejs.org/examples/webgl2_multiple_rendertargets.html`
3. `npm run tauri dev`
4. See error `Your browser does not seem to support WebGL 2`, no messages in the Web Inspector console.

### **Expected behavior**
See 3D interactive visualization, like: [webgl2 - multiple render targets](https://threejs.org/examples/webgl2_multiple_rendertargets.html)

### **Platform and Versions:**
```
➜  tauri-app npx tauri info
 app:spawn Running "~/work/_tmp/tauri-app/node_modules/@tauri-apps/cli/bin/tauri-cli tauri info" +0ms


Operating System - Mac OS, version 11.6.1 X64

Node.js environment
  Node.js - 14.16.0
  @tauri-apps/cli - 1.0.0-beta.10
  @tauri-apps/api - 1.0.0-beta.8

Global packages
  npm - 8.1.0
  yarn - 1.22.17

Rust environment
  rustc - 1.56.1
  cargo - 1.56.0

App directory structure
/dist
/node_modules
/src-tauri

App
  tauri.rs - 1.0.0-beta.8
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - ../dist
  devPath - ../dist

```

### **Additional context**
I tried to run three.js example by simply embedding it using iframe - same behavior.
WebGL example runs as expected in iframe.

**Comment by FabianLars** [comment]
webkit's webgl2 support is simply "too new". The system webkit version needs to be the one used in Safari 15 (which introduced webgl2 support).
Unfortunately updating safari doesn't update the system webkit :/
So i guess webgl2 only works on macOS >= v12 (not sure about that one)

**Comment by dawidgorny** [comment]
@FabianLars thanks for the answer! I was suspecting this sort of situation - I will try to test on another machine with macOS 12 later.

**Comment by FabianLars** [comment]
I'm gonna close this. Please reopen if it doesn't work on macOS 12 either.

**Comment by grschafer** [comment]
I also get the `Your browser does not seem to support WebGL 2` message when running https://threejs.org/examples/webgl2_multiple_rendertargets.html (I just opened the devtools in a tauri app and set `window.location` to that url).

My output for `npx tauri info`:

```
Environment
  › OS: Ubuntu 20.04 X64
  › Node.js: 18.2.0
  › npm: 8.9.0
  › pnpm: Not installed!
  › yarn: 1.22.19
  › rustup: 1.24.3
  › rustc: 1.60.0
  › cargo: 1.60.0
  › Rust toolchain: stable-x86_64-unknown-linux-gnu 

Packages
  › @tauri-apps/cli [NPM]: 1.0.1
  › @tauri-apps/api [NPM]: 1.0.1
  › tauri [RUST]: 1.0.1,
  › tauri-build [RUST]: 1.0.1,
  › tao [RUST]: 0.12.0,
  › wry [RUST]: 0.19.0,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: http://localhost:3000/
  › framework: Svelte
```

Since I'm on Ubuntu, I'm guessing webgl2 support is based on the webkit version bundled in the libwebkit2gtk package? My installed version is: 

```
libwebkit2gtk-4.0-37/focal-updates,focal-security,now 2.36.3-0ubuntu0.20.04.1 amd64
```

This might be the corresponding issue for adding webgl2 support to WebKitGTK: https://bugs.webkit.org/show_bug.cgi?id=166536 (there's a bit more explanation in [this duplicate issue](https://bugs.webkit.org/show_bug.cgi?id=235626)). From those issues it seems like webgl2 support is blocked or uncertain, unfortunately.

Let me know if this should be a separate linux-focused issue or if I'm missing something in the above!

**Comment by FabianLars** [comment]
> From those issues it seems like webgl2 support is blocked or uncertain, unfortunately.

Yep :/

> Let me know if this should be a separate linux-focused issue

No need as this is outside of our control, and since it's mentioned here now it should be easy enough to find.

**Comment by jon-chuang** [comment]
WebGL2 support now available in 2.39.1

https://webkitgtk.org/2022/11/11/webkitgtk2.39.1-released.html

Can this issue be reopened?

Seems that unfortunately even the unstable release for Debian has not caught up yet: https://packages.debian.org/unstable/libs/libwebkit2gtk-5.0-0

**Comment by FabianLars** [comment]
There is no action required on our side. once 2.40 releases it should be enabled by default and therefore available automatically.

No need to re-open the issue imo.
